ts/a11y/explorer/KeyExplorer.ts

```diff
@@ -253,31 +260,39 @@ export class SpeechExplorer extends AbstractKeyExplorer<string> {
   public Update(force: boolean = false) {
     super.Update(force);
     let options = this.speechGenerator.getOptions();
-    SRE.setupEngine({modality: options.modality,
-                     locale: options.locale});
-    this.region.Update(this.walker.speech());
-    // This is a necessary in case speech options have changed via keypress
-    // during walking.
-    if (options.modality === 'speech') {
-      this.document.options.sre.domain = options.domain;
-      this.document.options.sre.style = options.style;
-      this.document.options.a11y.speechRules =
-        options.domain + '-' + options.style;
-    }
+    SpeechExplorer.updatePromise.then(() => {
+      SpeechExplorer.updatePromise = new Promise((res) => {
+        Sre.sreReady()
+          .then(() => Sre.setupEngine({modality: options.modality,
+                                   locale: options.locale}))
+          .then(() => this.region.Update(this.walker.speech()))
+          .then(() => res());
+      });
```

Comment:
Similar to the discussion about the previous `updatePromise`, I think this needs to become

``` js
    SpeechExplorer.updatePromise = SpeechExplorer.updatePromise.then(async () => {
      return Sre.sreReady()
        .then(() => Sre.setupEngine({modality: options.modality,
                                     locale: options.locale}))
        .then(() => this.region.Update(this.walker.speech()));
    });
```

Note also that the lines below (271 to 279) will be done before the content of this promise is run, so it doesn't matter if those lines are before or after the setting of `updatePromise()`.  I think that is OK, but wanted to check with you to be sure.